src/App.tsx

```diff
@@ -1,26 +1,107 @@
+/* eslint-disable max-len */
+import React, { useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterTypes } from './types/FilterTypes';
+import { todosSlice } from './features/todos';
+import { currentTodoSlice } from './features/currentTodo';
+import { filterSlice } from './features/filter';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from './app/store';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+export const App: React.FC = () => {
+  const todos = useSelector((state: RootState) => state.todos);
+  const selectedTodo = useSelector((state: RootState) => state.currentTodo);
+  const query = useSelector((state: RootState) => state.filter.query);
+  const filterMethod = useSelector((state: RootState) => state.filter.status);
+
+  const dispatch = useDispatch();
+
+  const onClickHandler = (currentTodo: Todo | null) => {
+    dispatch(currentTodoSlice.actions.selectTodo(currentTodo));
+  };
+
+  useEffect(() => {
+    getTodos().then(todosFromServer => {
```

Comment:
Consider handling errors that might occur during the fetching of todos. You can do this by adding a `.catch` block to handle any potential errors. This will improve the reliability of your application.